Issue 3 - Added tests to check input/output format

redouane-dziri opened the pull request.

redouane-dziri committed:
added tests to check input/output format

redouane-dziri committed:
added new line at end of file

RoyalTS commented:
Those all look great! It's good practice though to have only a single test per test function as that makes it easier to pinpoint what tests are succeeding/failing. Would also be great to have variable names that are a bit more descriptive. If they're `wrong_probs` what exactly is wrong with them, etc.

RoyalTS commented on tests/test_stochatreat.py:
```diff
@@ -111,3 +111,121 @@ def test_stochatreat_only_misfits(probs):
+            treats=right_treats,
+            idx_col=idx_col,
+            probs=wrong_probs,
+            random_state=42,
```
would remove the random_state in these as it's non-essential for error behavior

redouane-dziri committed:
breaking tests into unit tests, refactored some fixtures, renamed variables

RoyalTS committed:
move tests around, add FIXMEs, make leaner

redouane-dziri committed:
corrected tests, and move the sorting after all necessary checks have been conducted

redouane-dziri committed:
refined test structure, added tests for output dtypes

redouane-dziri committed:
merge

redouane-dziri committed:
Update test_stochatreat.py

added a block id to the return df to retrieve within which blocks were the treatments assigned

removed unused variable

replaced code by simple pandas ngroup call

no more inplace operations

added test to check the right number of block ids was returned

minor syntax change

assigned column and drop old one

added a block id to the return df to retrieve within which blocks the treatments were assigned

added blank line at end of test file

added a block id to the return df to retrieve within which blocks were the treatments assigned, added testing to check the number of ids produced

moved sorting after all checks are done, and replaced syntax for filtering on columns to avoid problems if input has column named 'block'

added a block id to the return df to retrieve within which blocks were the treatments assigned, added corresponding tests, moved sorting after all checks are done, and replaced syntax for filtering on columns to avoid problems if input has column named 'block'

created input/output tests

redouane-dziri committed:
Merge branch 'master' into issue-3

RoyalTS commented:
The important bits of this are already on master so closing this.
